bindings/jupyroot/python/JupyROOT/helpers/utils.py

```diff
@@ -412,9 +412,47 @@ def _canJsDisplay(self):
 
 
     def _getJsCode(self):
+        # Add extra primitives to canvas with custom colors, palette, gStyle
```

Comment:
In order to encapsulate better the code, do you think you could create a helper method for the preprocessing? 
That helper method can be defined here in the `NotebookDrawer` class and should start its name with `_` to flag that it  is private (not intended to be invoked from outside the class). Also, it would be good to include some documentation of what it does, in a comment like this:
```
def _some_name_for_preprocessing_method(self):
''' Some documentation
for this method.
"""
    # code here
```
This way the `_getJsCode` function would look like:
```
def _getJsCode(self):
    style, colors, palette = self._some_preprocess_name()
    
    # Workaround to have ConvertToJSON work
    json = ROOT.TBufferJSON.ConvertToJSON(self.drawableObject, 3)
    
    # Cleanup primitives after conversion
    if style is not None: prim.Remove(style)
    if colors is not None: prim.Remove(colors)
    if palette is not None: prim.Remove(palette)

     ...
```

Reply:
Does it make sense of code used only once?

Reply:
Yes it does, it improves readabilty. You encapsulate code in a function with a given purpose, and prevent the main function from being too long.

Reply:
Just a tip: if in the helper function you do `return style, colors, palette` at the very end, it will put those three in a tuple that will be expanded in `_getJsCode`, with the code I wrote above.